source/beamline/SGM/SGMBypassLadder.cpp

```diff
@@ -0,0 +1,14 @@
+#include "SGMBypassLadder.h"
+
+SGMBypassLadder::SGMBypassLadder(const QString &name, const QString &baseName, QObject *parent) :
+	SGMLadderControl(name, baseName, parent)
+{
+	addValueOption(Slit, "Slit", SGMBYPASSLADDER_SLIT_SETPOINT, SGMBYPASSLADDER_SLIT_MIN, SGMBYPASSLADDER_SLIT_MAX);
```

Comment:
The first argument in addValueOption is an index for a particular option. I've defined enums for each of the ladder controls that describe the different options for that particular ladder. I use the enum here as the index value.